src/cmd/cli/command/commands.go

```diff
@@ -841,15 +845,20 @@ var composeUpCmd = &cobra.Command{
 			}
 		}()
 		if err := waitServiceStatus(ctx, cli.ServiceStarted, serviceInfos); err != nil && !errors.Is(err, context.Canceled) {
-			if !errors.Is(err, cli.ErrDryRun) && !errors.As(err, new(cliClient.ErrNotImplemented)) {
+			if errors.Is(err, ErrorDeploymentFailed) {
+				term.Warn("Deployment FAILED. Service(s) not running.")
```

Comment:
`return err` here. This would show the error message and exit the CLI with a non-zero exit code = win win